Packs/CybelAngel/Integrations/CybelAngelEventCollector/CybelAngelEventCollector.py

```diff
@@ -0,0 +1,289 @@
+    return latest_report_time, [report["id"] for report in reports if report["_time"] == latest_report_time]
+
+
+def test_module(client: Client) -> str:
+    """
+    Tests that the authentication to the api is ok.
+    """
+    client.get_reports(
+        start_date=(datetime.now() - timedelta(days=1)).strftime(DATE_FORMAT),
+        end_date=datetime.now().strftime(DATE_FORMAT)
+    )
+    return "ok"
+
+
+def fetch_events(client: Client, first_fetch: str, last_run: dict, max_fetch: int) -> tuple[List[dict[str, Any]], dict[str, Any]]:
+    """
+    Fetches reports from Cybel Angel
+
+    Args:
+        client: Cybel Angel client
+        first_fetch: since when to start to takes reports
+        last_run: the last run object
+        max_fetch: maximum number of reports
+
+    Fetch logic:
+    1. Get the latest report time from last fetch or start from fetch in case its a the first time fetching
+    2. get all the reports since the last fetch or first fetch
+    3. remove any reports which where already fetched
+    4. if there are no reports after dedup, keep the last run the same and return
+    5. if there are reports after dedup, update the last run to the latest report time, save all the report IDs which
+       occurred in the last event time
+    6. return all the fetched events
+
+    """
+    last_run_time = last_run.get(LastRun.LATEST_REPORT_TIME)
+    if not last_run_time:
+        last_run_time = dateparser.parse(first_fetch).strftime(DATE_FORMAT)  # type: ignore[union-attr]
+        if not last_run_time:
+            demisto.error(f'First fetch {first_fetch} is not valid')
+            raise ValueError(f'First fetch {first_fetch} not valid')
+    else:
+        last_run_time = dateparser.parse(last_run_time).strftime(DATE_FORMAT)  # type: ignore[union-attr]
+    now = datetime.now()
+    reports = client.get_reports(start_date=last_run_time, end_date=now.strftime(DATE_FORMAT), limit=max_fetch)
+    reports = dedup_fetched_events(reports, last_run_fetched_event_ids=last_run.get(LastRun.LATEST_FETCHED_REPORTS_IDS) or set())
+    if not reports:
+        demisto.debug(f'No reports found when last run is {last_run}')
+        return [], {
+            LastRun.LATEST_REPORT_TIME: last_run_time,
+            LastRun.LATEST_FETCHED_REPORTS_IDS: last_run.get(LastRun.LATEST_FETCHED_REPORTS_IDS)
+        }
+
+    latest_report_time, latest_fetched_report_ids = get_latest_event_time_and_ids(reports)
+    demisto.debug(f'latest-report-time: {latest_report_time}')
+    demisto.debug(f'latest-fetched-report-ids {latest_fetched_report_ids}')
+
+    last_run.update(
+        {
+            LastRun.LATEST_REPORT_TIME: latest_report_time,
+            LastRun.LATEST_FETCHED_REPORTS_IDS: latest_fetched_report_ids
+        }
+    )
+    return reports, last_run
+
+
+def get_events_command(client: Client, args: dict[str, Any]) -> CommandResults:
+    """
+    Get events from Cybel Angel, used mainly for debugging purposes
```

Comment:
@GuyAfik do you want to change this to 'Send events from CybelAngel to XSIAM. Used mainly for debugging.'

Reply:
i will update it yes, thanks